zebra-chain/src/serialization/date_time.rs

```diff
@@ -0,0 +1,102 @@
+//! DateTime types with specific serialization invariants.
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use chrono::{TimeZone, Utc};
+
+use std::{
+    convert::{TryFrom, TryInto},
+    fmt,
+    num::TryFromIntError,
+};
+
+use super::{SerializationError, ZcashDeserialize, ZcashSerialize};
+
+/// A date and time, represented by a 32-bit number of seconds since the UNIX epoch.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct DateTime32 {
+    timestamp: u32,
+}
+
+impl DateTime32 {
+    /// Returns the number of seconds since the UNIX epoch.
+    pub fn timestamp(&self) -> u32 {
+        self.timestamp
+    }
+
+    /// Returns the equivalent [`chrono::DateTime`].
+    pub fn as_chrono(&self) -> chrono::DateTime<Utc> {
```

Comment:
```suggestion
    pub fn to_chrono(&self) -> chrono::DateTime<Utc> {
```

Reply:
Clippy also wants `self` to be taken by value after the `to_chrono` change.

Fixed in a9a1e9c7